opm/material/fluidsystems/blackoilpvt/Co2GasPvt.cpp

```diff
@@ -125,34 +130,43 @@ setReferenceDensities(unsigned regionIdx,
     brineReferenceDensity_[regionIdx] = rhoRefBrine;;
 }
 
-template<class Scalar, class Params>
-void Co2GasPvt<Scalar, Params>::
+template<class Scalar, class Params, class ContainerT>
+OPM_HOST_DEVICE void Co2GasPvt<Scalar, Params, ContainerT>::
 setActivityModelSalt(int activityModel)
 {
     switch (activityModel) {
     case 1:
     case 2:
     case 3: activityModel_ = activityModel; break;
     default:
+#if OPM_IS_INSIDE_DEVICE_FUNCTION
+        assert(false && "The salt activity model options are 1, 2 or 3");
+#else
         OPM_THROW(std::runtime_error, "The salt activity model options are 1, 2 or 3");
+#endif
     }
 }
 
-template<class Scalar, class Params>
-void Co2GasPvt<Scalar, Params>::
+template<class Scalar, class Params, class ContainerT>
+OPM_HOST_DEVICE void Co2GasPvt<Scalar, Params, ContainerT>::
 setThermalMixingModel(int thermalMixingModel)
 {
     switch (thermalMixingModel) {
     // 0 = Use pure CO2 entalpy
     case 0:  gastype_ = Co2StoreConfig::GasMixingType::NONE; break;
     // 1 = Account for vapporized water in gas phase (Mass fraction)
     case 1:  gastype_ = Co2StoreConfig::GasMixingType::IDEAL; break;
-    default: OPM_THROW(std::runtime_error, "The thermal mixing model options are 0 and 1");
+    default:
+#if OPM_IS_INSIDE_DEVICE_FUNCTION
+        assert(false && "The thermal mixing model options are 0 and 1");
```

Comment:
How will an assertation failure behave when triggered inside a GPU thread?

Reply:
each thread encounter the assertion output that string if it fails. This will cause a lot of output and the application will terminate immediately